From: "Cho, Yu-Chen" <acho@suse.com>
Date: Wed, 16 Sep 2015 14:10:12 +0800
Subject: [PATCH] Remove support the device ID's from the code 0x15AC and
 0x15AE
Patch-mainline: Never, SUSE specific
References: bsc#922833 FATE#318521

Due to HW availability we should not have support for the 0x15ac device in SP1.
The 0x15ae support while upstream has not been tested on real HW yet and won't
for about another month.

The code, if any, that is in the driver already can remain and we just pull both
of the ID's from the driver so that the driver does not recognize these devices.
Acked-by: Ronciak, John <john.ronciak@intel.com>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_type.h |    2 --
 drivers/net/ethernet/intel/ixgbe/ixgbe_x550.c |   14 --------------
 2 files changed, 16 deletions(-)
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_type.h
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_type.h
@@ -77,9 +77,7 @@
 #define IXGBE_DEV_ID_X550T		0x1563
 #define IXGBE_DEV_ID_X550EM_X_KX4	0x15AA
 #define IXGBE_DEV_ID_X550EM_X_KR	0x15AB
-#define IXGBE_DEV_ID_X550EM_X_SFP	0x15AC
 #define IXGBE_DEV_ID_X550EM_X_10G_T	0x15AD
-#define IXGBE_DEV_ID_X550EM_X_1G_T	0x15AE
 #define IXGBE_DEV_ID_X550_VF_HV	0x1564
 #define IXGBE_DEV_ID_X550_VF		0x1565
 #define IXGBE_DEV_ID_X550EM_X_VF	0x15A8
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_x550.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_x550.c
@@ -50,19 +50,12 @@ static void ixgbe_setup_mux_ctl(struct i
 static s32 ixgbe_identify_phy_x550em(struct ixgbe_hw *hw)
 {
 	switch (hw->device_id) {
-	case IXGBE_DEV_ID_X550EM_X_SFP:
-		/* set up for CS4227 usage */
-		hw->phy.phy_semaphore_mask = IXGBE_GSSR_SHARED_I2C_SM;
-		ixgbe_setup_mux_ctl(hw);
-
-		return ixgbe_identify_module_generic(hw);
 	case IXGBE_DEV_ID_X550EM_X_KX4:
 		hw->phy.type = ixgbe_phy_x550em_kx4;
 		break;
 	case IXGBE_DEV_ID_X550EM_X_KR:
 		hw->phy.type = ixgbe_phy_x550em_kr;
 		break;
-	case IXGBE_DEV_ID_X550EM_X_1G_T:
 	case IXGBE_DEV_ID_X550EM_X_10G_T:
 		return ixgbe_identify_phy_generic(hw);
 	default:
@@ -1540,10 +1533,6 @@ static enum ixgbe_media_type ixgbe_get_m
 	case IXGBE_DEV_ID_X550EM_X_KX4:
 		media_type = ixgbe_media_type_backplane;
 		break;
-	case IXGBE_DEV_ID_X550EM_X_SFP:
-		media_type = ixgbe_media_type_fiber;
-		break;
-	case IXGBE_DEV_ID_X550EM_X_1G_T:
 	case IXGBE_DEV_ID_X550EM_X_10G_T:
 		 media_type = ixgbe_media_type_copper;
 		break;
@@ -1697,9 +1686,6 @@ mac_reset_top:
 		IXGBE_WRITE_REG(hw, IXGBE_HLREG0, hlreg0);
 	}
 
-	if (hw->device_id == IXGBE_DEV_ID_X550EM_X_SFP)
-		ixgbe_setup_mux_ctl(hw);
-
 	return status;
 }
 
